src/core/nixl_agent.cpp

```diff
@@ -1087,6 +1087,28 @@ nixlAgent::releaseXferReq(nixlXferReqH *req_hndl) const {
     return NIXL_SUCCESS;
 }
 
+nixl_status_t
+nixlAgent::createGpuXferReq(const nixlXferReqH &req_hndl, nixlGpuXferReqH *&gpu_req_hndl) const {
+    if (!req_hndl.engine) {
+        NIXL_ERROR << "Invalid request handle: engine is null";
+        return NIXL_ERR_INVALID_PARAM;
+    }
+
+    NIXL_SHARED_LOCK_GUARD(data->lock);
+    return req_hndl.engine->createGpuXferReq(req_hndl.backendHandle, gpu_req_hndl);
+}
+
+void
+nixlAgent::releaseGpuXferReq(const nixlXferReqH &req_hndl, nixlGpuXferReqH *gpu_req_hndl) const {
```

Comment:
didn't we want to remove req_hndl from function parameters?
though, i would requre smth like std::unordered_map from gpu_req_hndl to engine